### PR TITLE
Convert some framework classes to Kotlin

### PR DESCRIPTION
**Pull request by anarchuser**

### CODE REVIEW

File: sdk/src/framework/sc/shared/WinCondition.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package sc.shared
+
+import sc.api.plugins.ITeam
+
+/**
+ * Erzeugt eine neue Condition mit Sieger und Gewinngrund.
```
Comment (xerus2000) [inline sdk/src/framework/sc/shared/WinCondition.kt:6]: Was soll eine "Condition" sein?

**Review by xerus2000 — CHANGES_REQUESTED** [review]

File: sdk/src/framework/sc/shared/ScoreValue.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package sc.shared
+
+import com.thoughtworks.xstream.annotations.XStreamAlias
+import java.math.BigDecimal
+
+@XStreamAlias(value = "scoreValue")
+data class ScoreValue(val fragment: ScoreFragment, var value: BigDecimal)
```
[the remark this answers is not in the document]
Comment (anarchuser) [reply sdk/src/framework/sc/shared/ScoreValue.kt:7]: it had a setter before, though

File: sdk/src/framework/sc/shared/WinCondition.kt (hunk 1 above)
Comment (anarchuser) [reply sdk/src/framework/sc/shared/WinCondition.kt]: Keine Ahnung.
ist es mit 910e6c9 besser?

File: sdk/src/framework/sc/shared/ScoreValue.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (xerus2000) [reply sdk/src/framework/sc/shared/ScoreValue.kt:7]: yes, but we don't need to keep dirty architecture ;)

File: sdk/src/framework/sc/shared/WinCondition.kt (hunk 1 above)
Comment (xerus2000) [reply sdk/src/framework/sc/shared/WinCondition.kt]: ja :)

File: sdk/src/framework/sc/shared/ScoreValue.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (anarchuser) [reply sdk/src/framework/sc/shared/ScoreValue.kt:7]: I can see that, but then this whole thing isn't a refactor anymore

[the remark this answers is not in the document]
Comment (anarchuser) [reply sdk/src/framework/sc/shared/ScoreValue.kt]: cb8d600

[the remark this answers is not in the document]
Comment (anarchuser) [reply sdk/src/framework/sc/shared/ScoreValue.kt:7]: Actually, the GameRoomManager uses the setter.
So unless someone refactors this usage out, ScoreValue.value has to stay mutable

[the remark this answers is not in the document]
Comment (xerus2000) [reply sdk/src/framework/sc/shared/ScoreValue.kt:7]: yikes, I feel like I fixed that already somewhere else...

File: sdk/src/framework/sc/shared/ScoreFragment.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package sc.shared
+
+import com.thoughtworks.xstream.annotations.XStreamAlias
+import com.thoughtworks.xstream.annotations.XStreamAsAttribute
+
+@XStreamAlias(value = "scoreFragment")
+data class ScoreFragment @JvmOverloads constructor(
+        @XStreamAsAttribute
+        val name: String?,
```
Comment (xerus2000) [inline sdk/src/framework/sc/shared/ScoreFragment.kt:9]: i don't think this should be nullable

File: sdk/src/framework/sc/shared/ScoreFragment.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        val aggregation: ScoreAggregation? = ScoreAggregation.SUM,
```
Comment (xerus2000) [inline sdk/src/framework/sc/shared/ScoreFragment.kt:10]: neither this

**Review by xerus2000 — CHANGES_REQUESTED** [review]

File: sdk/src/framework/sc/shared/ScoreFragment.kt (hunk 3 above)
Comment (anarchuser) [reply sdk/src/framework/sc/shared/ScoreFragment.kt]: ah yeah, I missed those
ed28e0e9

**Review by xerus2000 — APPROVED** [review]

**Pull request merged**